pkg/odo/genericclioptions/runnable.go

```diff
@@ -120,7 +133,15 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 	}
 	util.LogErrorAndExit(err, "")
 
-	err = o.Run(cmdLineObj.Context())
+	if jsonOutputter, ok := o.(JsonOutputter); ok && log.IsJSON() {
```

Comment:
I can see in `genericclioptions.CheckMachineReadableOutputCommand` that when in json output, we force the option `v` to 0, but we don't modify the environment variable.
I think it would be nice to output the logs to a file when running with json output, like for other backend programs.
Let's work on this on another PR

Reply:
Created #5653 to keep track of this.